stellargraph/mapper/corrupted.py

```diff
@@ -96,31 +133,31 @@ def __getitem__(self, index):
 
         inputs, _ = self.base_generator[index]
 
-        if isinstance(
-            self.base_generator, (FullBatchSequence, SparseFullBatchSequence)
-        ):
+        shuffled_feats = []
+        for group in self.corrupt_index_groups:
+            feats_orig = [inputs[idx] for idx in group]
```

Comment:
completely optional, but the code in this inner loop might be well suited to being in a private function `_merge_shuffle_split(feats)`

Reply:
That's a good idea. I think it has to be a bit different because it's a "flat map", with multiple outputs per input group, not just a "map". Specifically, a pair of nested loops:

```python
        shuffled_feats = [
            corrupted
            for group in self.corrupt_index_groups
            for corrupted in corrupt_group(group)
        ]
```

Other than that, it does seem nicer.